### ISSUE TITLE
Valhalla is not usable from the build directory

### ISSUE THREAD

**Issue by mwoehlke-kitware** [valhalla/valhalla#2440]
I want to use Valhalla in another application. Unfortunately, I cannot just write:

```cmake
find_package(valhalla REQUIRED)
```

...because Valhalla is only made available via pkg-config.

It would be much more convenient if Valhalla a) provided CMake exported targets and b) was usable from the build directory, not just an installation.

**Comment by kevinkreiser** [comment]
@mwoehlke-kitware thats right, the install target does the typical linux install procedure of adding a pc file and putting the required stuff in `/usr/local/*`. i supose it would be pretty easy for us to write a `.cmake` file to let your project search for an installed version of valhalla.

i didnt understand the part about being usable from the build directory. you are saying you want your code to find the valhalla cmake build directory of an already built code base and use that? do  you have any examples of other projects doing this so i can get a clearer picture of exactly whta is meant? in my experience something similar would be to vendor valhalla as a submodule and then get access to its targets via `add_subdirectory(thirdparty/valhalla)`. this will also build valhalla as part of your project of course which may or may not be something you want.

**Comment by mandeepsandhu** [comment]
I'll let @mwoehlke-kitware clarify, but I think his ask is to be able to use valhalla in another project without having to install it (hence "usable from the build directory"). `cmake` provides the [export function](https://cmake.org/cmake/help/latest/command/export.html) which could be used to enable other projects to find & use valhalla's build directory as a dependency. I have to read more about `export` but it looks like `export(PACKAGE <PackageName>)` is what we need (this registers the build tree in cmake's registry, so other projects can use `find_package` to find it).

**Comment by mwoehlke-kitware** [comment]
> you are saying you want your code to find the valhalla cmake build directory of an already built code base and use that?

Yes.

> do you have any examples of other projects doing this so i can get a clearer picture of exactly whta is meant?

Most projects that provide exported targets will provide them for both the build and install trees. See for example most projects (ones that provide C++ libraries, anyway) under [Kitware](/Kitware). The simplest example that comes to mind offhand would be [qtExtensions](/Kitware/qtextensions). I believe most KDE libraries do this also. Other examples: GDAL, PDAL, GeographicLib, yaml-cpp, VXL... I'm sure there are many others; these are libraries for which I currently have builds and can therefore check easily.

I may have a crack at this, but I'm still waiting to find out if our customer allows us to contribute [things other than obvious bug fixes] without jumping through hoops.
